wandb/sdk/wandb_init.py

```diff
@@ -485,6 +496,12 @@ def init(self) -> Union[Run, RunDisabled, None]:  # noqa: C901
             with telemetry.context(run=run) as tel:
                 tel.feature.offline = True
             run_proto = backend.interface._make_run(run)
+
+            if s.resume is not None:
+                self._error_out(
+                    backend, error_message="Offline mode does not support resume"
```

Comment:
If we did the check before we created the backend, we wouldnt have to do the backend.cleanup() teardown code.
(that code is fraught with cases it doesn't handle properly)

Reply:
yeah i was trying to figure out if I should do it as soon as possible. 
Seems like the answer is yes. Let me move it up